components/Panel/Panel.tsx

```diff
@@ -51,57 +48,72 @@ export function Panel() {
   return (
     <>
       {transitions(
-        ({ opacity, right }, isOpen) =>
+        ({ opacity }, isOpen) =>
           isOpen && (
             <Overlay
-              onDismiss={() => closePanel(true)}
+              onClick={() => closePanel()}
               style={{
                 backgroundColor: opacity.to(
                   (value) => `hsla(280, 4%, 15%, ${value})`
                 ),
               }}
-            >
-              <Content
-                ref={contentRef}
-                aria-labelledby="panel-title"
-                style={
-                  {
-                    "--right": right.to((value) => `${value}px`),
-                    "--panel-width": `${panelWidth}px`,
-                  } as CSSProperties
-                }
-              >
-                <PanelComponent content={panelContent} />
-                <CloseButton
-                  onClick={() => closePanel()}
-                  style={
-                    {
-                      "--fill": closeButtonColor,
-                    } as CSSProperties
-                  }
-                >
-                  <IconWrapper>
-                    <CloseIcon />
-                  </IconWrapper>
-                </CloseButton>
-              </Content>
-            </Overlay>
+            ></Overlay>
           )
       )}
+      <FocusOn
```

Comment:
This component addresses the main accessibility concerns of a dialog like this by focusing the content and locking scroll of the document body.